705.py

```diff
@@ -0,0 +1,50 @@
+# N.B.
+# We should go back visit linkedlist and BST solutions.
+# We should do time complexity analysis for each HashSet function.
+
+class MyHashSet:
+
+    def __init__(self):
+        """
+        Initialize your data structure here.
+        """
+        self.SPACE_SIZE = 9999
+        self.space = [[] for _ in range(self.SPACE_SIZE)]
+
+    def add(self, key: int) -> None:
+        hashcode = self.computeHash(key)
```

Comment:
Related to what I just commented [here](https://github.com/bunnygod/leetcode-python/pull/149#discussion_r660698239), this variable does not mean or represent a **_hash_** or **_hash code_**; it means a location within the `self.space` array. 

A _**hash**_ is a value calculated based on one or more properties of a `key` object, and it can be obtained by calling `hash(a_key_object)` in Python. We then use this hash value to derive a location in the `self.space` array to store the `key` object; this location is what we need here to store the `key`. Therefore, there are two steps involved:

1. Compute the hash code of the `key` object
2. Map the hash code to a location in the `self.space` array

I want to emphasize the difference between a hash and a location in the array of buckets: we rely on the location index to identify the correct slot in the array of buckets to store the `key` object; we do not use an object's hash code to identify this slot. 

Currently, you are mixing these two concepts together. Your current implementation works only because your key object happens to be an `int` object whose value _can_ be used as a hash code. Imagine if your hash set implementation has to deal with objects of the `Animal` type as the keys, can you still use `dog % self.SPACE_SIZE` to get your location index? No, you will have to get the hash code of `dog` using `hash(dog)` first, and then map this hash code to a location index. What you return to `add()`, `remove()`, and `contains()` should always be a location index, not a hash code.